cli/src/templater.rs

```diff
@@ -75,13 +76,41 @@ impl Template for Signature {
         }
         if !self.email.is_empty() {
             write!(formatter, "<")?;
-            write!(formatter.labeled("email"), "{}", self.email)?;
+            let email: Email = self.email.clone().into();
+            email.format(formatter)?;
             write!(formatter, ">")?;
         }
         Ok(())
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Email {
+    pub local: String,
+    pub domain: Option<String>,
+}
```

Comment:
nit: might be simpler to implement as a new type `Email(pub String)` (because the type implements implicit cast to boolean based on string)

Reply:
Does this mean performing the splitting to local/domain on demand in the `Template::format` function and the `Email.local()` and `Email.domain()` function?

Reply:
Yes. Since `split_email()` never fails, `Email` doesn't have to be parsed at the construction.